app/App/Account/Account/Requests/TransactionRequest/TxAction/index.js

```diff
@@ -89,6 +91,80 @@ class TxSending extends React.Component {
                 </ClusterRow>
               ) : null}
 
+              {address && (
+                <ClusterRow>
+                  <ClusterValue pointerEvents={true} onClick={() => {
+                    this.copyAddress(address)
+                  }}>
+                    <div className='clusterAddress'>
+                      {ensName
+                        ? <span className='clusterAddressRecipient'>{ensName}</span>
+                        : <span className='clusterAddressRecipient'>{address.substring(0, 8)}{svg.octicon('kebab-horizontal', { height: 15 })}{address.substring(address.length - 6)}</span>
+                      }
+                      <div className='clusterAddressRecipientFull'>
+                        {this.state.copied ? (
+                          <span>{'Address Copied'}</span>
+                        ) : (
+                          <span className='clusterFira'>{address}</span>
+                        )}
+                      </div>
+                    </div>
+                  </ClusterValue>
+                </ClusterRow>
+              )}
+            </Cluster>
+          </ClusterBox>
+        )
+      } else if (actionType === 'approve') {
+        const { amount, decimals, name, spender: recipientAddress, symbol, spenderType, spenderEns } = action.data || {}
+        const address = recipientAddress
+        const ensName = spenderEns
```

Comment:
any reason why we're assigning these?  looks like we can just do, e.g.

```
const { amount, decimals, name, spender: address, symbol, spenderType, spenderEns: ensName } = action.data || {}
```

and update the recipient below to use address